### ISSUE TITLE
Seems not work for skinmesh

### ISSUE THREAD

**Issue by yurilin** [Shrimpey/Outlined-Diffuse-Shader-Fixed#1]
I generated some mesh(polygon) by the script and set the material shader to them.
Unfortunately, they did not work.

I tested on a cube, it worked.

**Comment by Shrimpey** [comment]
I’ll have a look at it today or tomorrow, haven’t looked into them for a while so it may have been caused by an update. Also take note, that most of the shaders in the package are experimental and are expected to work strangly in certain conditions. Please provide information about your Unity version and the exact shader you,ve been using.

**Comment by yurilin** [comment]
Hi, thanks for your the replay.
I mean the mesh which did not have a volume but just a plane
like:
`var width: float;
var height: float;

function Start() {  
    var mf: MeshFilter = GetComponent.<MeshFilter>();
    var mesh = new Mesh();
    mf.mesh = mesh;
    
    var vertices: Vector3[] = new Vector3[4];
    
    vertices[0] = new Vector3(0, 0, 0);
    vertices[1] = new Vector3(width, 0, 0);
    vertices[2] = new Vector3(0, height, 0);
    vertices[3] = new Vector3(width, height, 0);
    
    mesh.vertices = vertices;
    
    var tri: int[] = new int[6];

    tri[0] = 0;
    tri[1] = 2;
    tri[2] = 1;
    
    tri[3] = 2;
    tri[4] = 3;
    tri[5] = 1;
    
    mesh.triangles = tri;
    
    var normals: Vector3[] = new Vector3[4];
    
    normals[0] = -Vector3.forward;
    normals[1] = -Vector3.forward;
    normals[2] = -Vector3.forward;
    normals[3] = -Vector3.forward;
    
    mesh.normals = normals;
    
    var uv: Vector2[] = new Vector2[4];

    uv[0] = new Vector2(0, 0);
    uv[1] = new Vector2(1, 0);
    uv[2] = new Vector2(0, 1);
    uv[3] = new Vector2(1, 1);
    
    mesh.uv = uv;
}`

the material is not important.I can try any which just can show the texture would be ok.  
but I don't know how to write an outline on such generated **flat** meshes.
Only I found on asset store was camera shaders so would not do a really wrap to each object.
If your script is possible, that will be great. If not, just fine.

**Comment by Shrimpey** [comment]
Alrighty, I'll have a look into it and try to experiment with some stuff. I'll let you know if I get it working :)

**Comment by Shrimpey** [comment]
_EDIT: I think the problem mostly lies in the way that mesh is created, "CustomOutline" seems to be working fine without tuning it, but with a different mesh generation._

Okay, changed two simple things in "UniformOutline", I'll get to fixing others in a sec. It seems to be working now, however I did not manage to get it working with your code, vertices seem strange. This is plane generation that it worked fine with (also added an example unitypackage to the repo, so you can download it and import to a project to test for yourself):

using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class temp : MonoBehaviour {
	public float width;
	public float height;

	void Start() {
		MeshFilter mf = GetComponent<MeshFilter>();
		mf.mesh = CreateMesh (width, height);
	}


	Mesh CreateMesh(float w, float h)
	{
		Mesh m = new Mesh();
		m.name = "ScriptedMesh";
		m.vertices = new Vector3[] {
			new Vector3(-w, -h, 0),
			new Vector3(w, -h, 0),
			new Vector3(w, h, 0),
			new Vector3(-w, h, 0)
		};
		m.uv = new Vector2[] {
			new Vector2 (0, 0),
			new Vector2 (0, 1),
			new Vector2(1, 1),
			new Vector2 (1, 0)
		};
		m.triangles = new int[] { 0, 1, 2, 0, 2, 3};
		m.RecalculateNormals();

		return m;
	}
}

_EDIT: Also updated "RegularOutline" shader. "CustomOutline" seems to be working fine out of the box._

**Comment by yurilin** [comment]
Hi, I tried the test unitypackage.
seems the outline not complete for a triangle like thing
	void Start() {
		MeshFilter mf = GetComponent<MeshFilter>();
		mf.mesh = CreateMesh (width, height);
	}


	Mesh CreateMesh(float w, float h)
	{
		Mesh m = new Mesh();
		m.name = "ScriptedMesh";
        m.vertices = new Vector3[] {
            new Vector3(-w, -h, 0),
            new Vector3(w, -h, 0),
            new Vector3(w, h, 0)
			//new Vector3(-w, h, 0)

        };
		m.uv = new Vector2[] {
			new Vector2 (0, 0),
			new Vector2 (0, 1),
			new Vector2(1, 1)
			//new Vector2 (1, 0)
		};
		m.triangles = new int[] { 0, 1, 2};
		m.RecalculateNormals();

		return m;

This would not draw the hypotenuse's outline of the triangle.
 
And one more thing, I found the mesh seems 180 y rotated.

**Comment by Shrimpey** [comment]
So the triangular thing is connected to the way that the shader works (scaling the vertices), a simple fix is just not possible here. However, I've added a separate shader just for triangles, so if your need is very specific you can just use this shader (triangleOutline.shader). In other cases you should pick a different one. 

You can also manually tweak the triangle shader as it is quite a rough solution, you can mess around with:
`v.vertex.x -= _OutlineWidth/4;
v.vertex.y += _OutlineWidth/4;`
in lines 48 and 49.

**Comment by yurilin** [comment]
Hey, that's very kind of you. 
My goal is draw outline to any shape of flat polygons(generated by script like the upon way), but not a just specific type.
This shader I think is very useful for 2d games(ex. jigsaw games) although there are sprite outline shaders, but mesh polyon is completely the different.
Anyway, I did not mean I demand or require you to upgrade the shader. Sorry for wasting your time these days.XD
I just want to give such a suggestion because seems there's no one on earth has made such a shader which could draw a perfect outline on flat meshes.

You don't have to do it unless you feel have the time or necessary.
Thanks again!

**Comment by Shrimpey** [comment]
Nah, I didn’t waste no time with that, I always find it interesting to mess around with shaders, I ain’t no expert with them, but I enjoy experimenting nevertheless. Since I started working with them I found that there is no reliable way to achieve a universal outline shader, so I am afraid that it may be the case with flat shapes too. However, I believe that it is a simplier case, so maybe after starting from the ground up there is a chance to make a perfect shader for this scenario. I may do some experimenting with this in the near future.

**Comment by Shrimpey** [comment]
Also, after reading some stuff online, I found that making such a shader would require designing a Geometry shader (which would require directx11+ to run), because it needs to scale triangles, a vertex shader can only work with vertices.
